src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonFinderFeatures.kt

```diff
@@ -1,189 +1,376 @@
-                event.stackTip = itemName.split(' ').last().romanToDecimalIfNecessary().toString()
-            }
-        } else if (itemName.startsWith("The Catacombs - ") || itemName.startsWith("MM Catacombs -")) {
-            val floor = itemName.split(" - ").last().removeColor()
-            val floorNum = floor.split(' ').last().romanToDecimalIfNecessary().toString()
-            val isMasterMode = itemName.contains("MM ")
-
-            event.stackTip = if (floor.contains("Entrance")) {
-                "E"
-            } else if (isMasterMode) {
-                "M${floorNum}"
-            } else {
-                "F${floorNum}"
-            }
-        } else if (itemName.endsWith("'s Party")) {
-            val floor = event.stack.getLore().find { it.startsWith("§7Floor: ") } ?: return
-            val dungeon = event.stack.getLore().find { it.startsWith("§7Dungeon: ") } ?: return
-            val floorNum = floor.split(' ').last().romanToDecimalIfNecessary().toString()
-            val isMasterMode = dungeon.contains("Master Mode")
-
-            event.stackTip = if (floor.contains("Entrance")) {
-                "E"
-            } else if (isMasterMode) {
-                "M${floorNum}"
-            } else {
-                "F${floorNum}"
-            }
+            stackTip(event)
+
+            highlightingHandler(event)
+
+            toolTipHandler(event)
         }
-    }
 
-    @SubscribeEvent
-    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
-        if (!LorenzUtils.inSkyBlock || LorenzUtils.skyBlockArea != "Dungeon Hub") return
-        if (event.inventoryName != "Catacombs Gate") return
+    }
 
-        val lore = event.inventoryItems[45]?.getLore() ?: return
+    private fun stackTip(event: InventoryFullyOpenedEvent) {
+        val inventoryName = event.inventoryName
+        if (catacombsGatePattern.matches(inventoryName)) catacombsGateStackTip(event)
+        if (!config.floorAsStackSize) return
+        if (selectFloorPattern.matches(inventoryName)) selectFloorStackTip(event)
+        if (partyFinderTitlePattern.matches(inventoryName)) partyFinderStackTip(event)
+    }
 
-        if (lore[0] == "§7View and select a dungeon class.") {
-            selectedClass = lore[2].split(" ").last().removeColor()
+    private fun selectFloorStackTip(event: InventoryFullyOpenedEvent) {
+        inInventory = true
+        for ((slot, stack) in event.inventoryItems) {
+            val name = stack.displayName.removeColor()
+            if (anyFloorPattern.matches(name)) {
+                floorStackSize[slot] = "A"
+            } else if (entranceFloorPattern.matches(name)) {
+                floorStackSize[slot] = "E"
+            } else if (floorPattern.matches(name)) {
+                floorStackSize[slot] =
+                    floorNumberPattern.matchMatcher(name) { group("floorNum").romanToDecimalIfNecessary().toString() }
+                        .toString()
```

Comment:
I don't see the reason to do .toString on this a second time, maybe just ?: continue if its null